### ISSUE TITLE
Table teampass.users doesn't exist

### ISSUE THREAD

**Issue by AFEIJIAN** [nilsteampassnet/TeamPass#2720]
### Steps to reproduce
1. Install teampass into web root as usual
2. Specify required credentials via Teampass installation page
3. Go to homepage via `Move to home page` link in the end of installation page

### Expected behaviour
Login page should be shown

### Actual behaviour
Error show up: <br>
`Table 'pwd_teampass.users' doesn't exist16fHXQsLCwsG3Koe0KiqDMQ2tW5vbkTA68zn2qDg6hl2f0u6c+FO+QQgwLAu7LKgyVHYjMf+ry7Gi7syu1q0/oiA0FvLvy5XQIAu6/dsXE0e3Xq7Za88ONTIvvij3NeIZu3M/xA=` where `pwd_teampass` is my mysql database name

### Server configuration
**Operating system**: Raspbian Buster Lite (based on `Debian Buster`)

**Web server:** Apache2

**Database:** MySQL

**PHP version:** 7.1

**Teampass version:** 2.1.27.27

**Teampass configuration file:** `includes/config/settings.php`

**Updated from an older Teampass or fresh install:** fresh install
PLEASE attach to this issue the file `/includes/config/tp.config.php`.
[tp.config.php.txt](https://github.com/nilsteampassnet/TeamPass/files/3828124/tp.config.php.txt)

### Client configuration
**Browser:** Google Chrome 78.0.3904.87

**Operating system:** Windows 10

#### Web server error log
```
[Sun Nov 10 13:40:36.790523 2019] [proxy_fcgi:error] [pid 18394:tid 3023029280] (70007)The timeout specified has expired: [client 192.168.8.101:57476] AH01075: Error dispatching request to : (polling)
[Sun Nov 10 13:41:16.123313 2019] [proxy_fcgi:error] [pid 18394:tid 3048207392] [client 192.168.8.101:57532] AH01071: Got error 'PHP message: PHP Notice:  Undefined index: key in /var/www/production/pwd/sources/main.functions.php on line 1429\n'
[Sun Nov 10 13:43:03.431114 2019] [ssl:warn] [pid 18391:tid 3069419232] AH01906: 192.168.8.198:443:0 server certificate is a CA certificate (BasicConstraints: CA == TRUE !?)
[Sun Nov 10 13:43:12.432343 2019] [ssl:warn] [pid 18391:tid 3069419232] AH01906: 192.168.8.198:443:0 server certificate is a CA certificate (BasicConstraints: CA == TRUE !?)
[Sun Nov 10 13:51:59.275014 2019] [proxy_fcgi:error] [pid 25659:tid 2970600480] [client 192.168.8.101:57679] AH01071: Got error 'Primary script unknown\n'
[Sun Nov 10 13:52:00.644839 2019] [proxy_fcgi:error] [pid 25659:tid 3048207392] [client 192.168.8.101:57679] AH01071: Got error 'Primary script unknown\n'
[Sun Nov 10 13:52:03.600534 2019] [proxy_fcgi:error] [pid 25659:tid 3006243872] [client 192.168.8.101:57679] AH01071: Got error 'PHP message: PHP Notice:  Undefined index: key in /var/www/production/pwd/sources/main.functions.php on line 1429\n'
[Sun Nov 10 14:00:45.955695 2019] [proxy_fcgi:error] [pid 25660:tid 2947494944] [client 192.168.8.101:57877] AH01071: Got error 'PHP message: PHP Notice:  Undefined index: key in /var/www/production/pwd/sources/main.functions.php on line 1429\n', referer: https://192.168.8.198/pwd/install/install.php
[Sun Nov 10 14:01:01.411905 2019] [proxy_fcgi:error] [pid 25659:tid 3039814688] [client 192.168.8.101:57885] AH01071: Got error 'Primary script unknown\n'
[Sun Nov 10 14:01:59.136213 2019] [proxy_fcgi:error] [pid 25659:tid 2937029664] [client 192.168.8.101:57919] AH01071: Got error 'PHP message: PHP Notice:  Undefined index: key in /var/www/production/pwd/sources/main.functions.php on line 1429\n'

```

#### Log from the web-browser developer console (CTRL + SHIFT + i)
(empty)

**Comment by ozawa-hi** [comment]
So, does the table teampass_users exist in the pwd_teampass scheme?
If it doesn't, there probably was an error during the setup. Did you specify the database user/password and connection information correctly?

**Comment by AFEIJIAN** [comment]
No the table doesn't exist on it. But when I check via `phpmyadmin` it still created other tables. The database connection information I had given are correct and I even connected it manually.

**Comment by ozawa-hi** [comment]
Seems like the installation didn't go correctly because table pwd_teampass.users should have been created during TeamPass setup.
During the setup, there's a step on "Table creation". Did it show table users being created?

e.g. setup steps.
http://www.techspacekh.com/installing-teampass-collaborative-passwords-manager-on-rhel-7centos-7/

**Comment by AFEIJIAN** [comment]
Unfortunately, it doesn't show up.

**Comment by Jatzoo** [comment]
The problem is related to this code from create table users (file install.queries.php):
```
(...)
`public_key` TEXT NOT NULL DEFAULT 'none',
`private_key` TEXT NOT NULL DEFAULT 'none',
```
If You try to do it yourself, the error are like this:
`
ERROR 1101 (42000): BLOB/TEXT column 'public_key' can't have a default value,
ERROR 1101 (42000): BLOB/TEXT column 'private_key' can't have a default value,
`

This type of errors are not presented to user during Teampass installation process.

**Comment by AFEIJIAN** [comment]
I see, looks like this is a bug?

**Comment by cummie** [comment]
Hi,
I found a point where the symptoms changes:
In my first scenario I used CentOS 7.7 with minimal installation and after finishing the teampass setup I got this error Table 'teampass.teampass_users' doesn't exist.
Then I tried the same with CentOS 8.0 and now I am a step further.
I get the login screen, but I can not login with the admin user. 
Kindly regards

**Comment by b-kannan** [comment]
Hi,
This error is seen on Centos 8 installation. The error message is as follows:

Table 'teampass.teampass_users' doesn't existulRMXldXV1elPXEH0ZIoDCwAEdWzjLVNHlkcPldxvgRDSpOecLhxGfdG5XCA8fz1BG9ueG/Z1jnr5iYPwO6UxTvRx589GuYyG/mTJZf60sJmRJ3I5wM9lXs3RZYz0cyeoZZUIaj5ekXiMw==

The table is not being created and there are no errors shown during the installation process. Any quick fixes?

Thanks

**Comment by moltanov** [comment]
Hi,

I found a fix for this issue, I manually created the users table by copying the SQL query in install.queries.php (located in teampass/install) and launching it in phpmyadmin. The part that seemed to cause the issue for me was the default value for `user_ip_lastdate,` I changed `NULL `to `'none'`. Afterward I had to redo the installing process by deleting the files in apache's directory and reputting the one I downloaded, I think changing the install.query file before install should also work.

I used Debian 10 with Apache and a mysql db.

Regards,
